### DISCUSSION TITLE
Making the most out of select component

### DISCUSSION THREAD

**Discussion by amirsafaricg** [quasarframework/quasar#11893] [General - Components / Directives / etc] [answered]
Hi guy's 

I wanted to combine the customizing menu option and the filtering and autocomplete in the select docs 
but I can't 

can anyone guide me through ?

**Comment by pdanpdan** [comment]
Show something you have tried and it's not working

**Comment by amirsafaricg** [comment]
I dont know that could be useful 
let me explain where my issue is 

while customizing the menu we need an array made out of objects 
like this : 

> options: [
        {
          label: 'Google',
          value: 'Google',
          description: 'Search engine',
          icon: 'mail'
        },
        {
          label: 'Facebook',
          value: 'Facebook',
          description: 'Social media',
          icon: 'bluetooth'
        },
        {
          label: 'Twitter',
          value: 'Twitter',
          description: 'Quick updates',
          icon: 'map'
        },
        {
          label: 'Apple',
          value: 'Apple',
          description: 'iStuff',
          icon: 'golf_course'
        },
        {
          label: 'Oracle',
          value: 'Oracle',
          disable: true,
          description: 'Databases',
          icon: 'casino'
        }
      ]


and the example for filtering is using this code to filter based on what we enter like this : 

> filterFn (val, update) {
        if (val === '') {
          update(() => {
            options.value = stringOptions

            // here you have access to "ref" which
            // is the Vue reference of the QSelect
          })
          return
        }

        update(() => {
          const needle = val.toLowerCase()
          options.value = stringOptions.filter(v => v.toLowerCase().indexOf(needle) > -1)
        })
      }



now my problem is that the example used by the docs in filtering is using an array just made out of some strings 
but while customizing our menu, we have to use an array made out of objects 

how can I make a select component with customized menu and also filtering applied ?

**Comment by MilosPaunovic** [comment, marked as answer]
Is this what are you looking for:
[https://codepen.io/milospaunovic/pen/gOGKMPR?editors=1011](https://codepen.io/milospaunovic/pen/gOGKMPR?editors=1011)

Please, next time, do create a Codepen of your own so it would be easier for someone to help you.

**Reply by amirsafaricg** [reply]
Thank you of course thanks again :)
